libcaf_net/caf/net/lp/framing.cpp

```diff
@@ -42,12 +42,27 @@ class framing_impl : public framing {
 
   static constexpr size_t hdr_size = sizeof(uint32_t);
 
-  static constexpr size_t max_message_length = INT32_MAX - sizeof(uint32_t);
-
   // -- constructors, destructors, and assignment operators --------------------
 
-  explicit framing_impl(upper_layer_ptr up) : up_(std::move(up)) {
-    // nop
+  explicit framing_impl(upper_layer_ptr up, dsl::size_field_t lp_size)
```

Comment:
`explicit` only makes sense on single argument constructors (or constructors with default parameters that may be called with one argument).

```suggestion
  framing_impl(upper_layer_ptr up, dsl::size_field_t lp_size)
```